base/version.jl

```diff
@@ -80,8 +79,8 @@ struct VersionNumber
     end
 end
 VersionNumber(major::Integer, minor::Integer = 0, patch::Integer = 0,
-        pre::Tuple{Vararg{Union{Integer,AbstractString}}} = (),
-        bld::Tuple{Vararg{Union{Integer,AbstractString}}} = ()) =
+              @nospecialize(pre::Tuple{Vararg{Union{Integer,AbstractString}}} = ()),
+              @nospecialize(bld::Tuple{Vararg{Union{Integer,AbstractString}}} = ())) =
```

Comment:
On second thought, this is probably not the right change...

We'd really like to specialize on the `Vararg` parameters here, but not on the length of the tuple.

Reply:
For now I'm going to back out of this part of these changes, until we have a solution for: https://github.com/JuliaLang/julia/issues/54390

Thankfully, the other call-site should be fine to de-specialize since it's a small, concrete union.